test/jdk/javax/swing/JPopupMenu/bug4234793.java

```diff
@@ -0,0 +1,255 @@
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/* @test
+ * @bug 4234793
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @summary PopupMenuListener popupMenuCanceled is never called
+ * @run main/manual bug4234793
+ */
+
+import java.awt.BorderLayout;
+import java.awt.Component;
+import java.awt.Dimension;
+import java.awt.event.ActionEvent;
+import java.awt.event.InputEvent;
+import java.awt.event.KeyEvent;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+
+import javax.swing.AbstractAction;
+import javax.swing.JButton;
+import javax.swing.JComboBox;
+import javax.swing.JFrame;
+import javax.swing.JLabel;
+import javax.swing.JMenu;
+import javax.swing.JMenuBar;
+import javax.swing.JMenuItem;
+import javax.swing.JPanel;
+import javax.swing.JPopupMenu;
+import javax.swing.JSeparator;
+import javax.swing.KeyStroke;
+import javax.swing.event.PopupMenuEvent;
+import javax.swing.event.PopupMenuListener;
+
+/**
+ * For all 3 components (JPopupMenu, JComboBox, JPopup) when the popup is visible,
+ * the popupMenuCanceled should be invoked in these two circumstances:
+ *
+ * 1. The ESCAPE key is pressed while the popup is open.
+ *
+ * 2. The mouse is clicked on another component.
+ *
+ * @test
+ *
+ */
+public class bug4234793 extends JFrame implements PopupMenuListener {
+
+    static final String INSTRUCTIONS = """
+        The test window will contain several kinds of menus.
+
+        * A menu bar with two menus labeled "1 - First Menu" and "2 - Second Menu"
+        * A drop down combo box - ie a button which pops up a menu when clicked
+        * Clicking any where on the background of the window will display a popup menu
+
+        That is 4 menus in total.
+
+        For each case, verify that the menu can be cancelled (hidden) in two ways
+        1) Click to display the menu, then to hide it, press the ESCAPE key.
+        2) Click to display the menu, then to hide it, LEFT click on the window background.
+        Note : the popup menu must be displayed using RIGHT click, the others use LEFT click.
+
+        Notice each time you perform a hide/cancel action an appropriate message should
+        appear in the log area
```

Comment:
_"Click to display the menu, then to hide it, LEFT click on the window background."_ does not produce PopupMenu cancelled msg for the 2 menus on the menubar and the combobox menu. But clicking the mouse on another component does cause the popup cancelled msg to be displayed.

Reply:
Oh , I see why the msg is not logged in PFJ.logArea. 
The "popupMenuWillBecomeInvisible: " needs to be added to the PFJ.log() instead of standard out.